db/migrate/20170704102536_use_deleted_in_containers_tables.rb

```diff
@@ -0,0 +1,55 @@
+class UseDeletedInContainersTables < ActiveRecord::Migration[5.0]
+  class ContainerDefinition < ActiveRecord::Base
+  end
+
+  class ContainerGroup < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  class ContainerImage < ActiveRecord::Base
+  end
+
+  class ContainerProject < ActiveRecord::Base
+  end
+
+  class ContainerNode < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  class Container < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  def disconnect_to_soft_delete(model)
+    model.where.not(:deleted_on => nil).update_all("ems_id = old_ems_id")
```

Comment:
so doing this, before doing  update_all("ems_id = old_ems_id")
```
    model.where.not(:deleted_on => nil).where.not(:ems_id => nil).update_all(:deleted_on => nil)
```

so having :ems_id will mean the record is not disconnected